lib/puppet/functions/parsepson.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+#
+# parsepson.rb
+#
+# @summary
+#   This function accepts PSON, a Puppet variant of JSON, as a string and converts
+#   it into the correct Puppet structure
+#
+# @example How to parse pson
+#   $data = parsepson("{any valid pson: string}")
```

Comment:
Is this valid pson? Looks more like hocon.

Reply:
My bad, copied the wrong code.
Took from the hocon test